src/agent/communicator/src/http_socket.cpp

```diff
@@ -0,0 +1,246 @@
+#include "http_socket.hpp"
+
+#include "http_client_utils.hpp"
+#include <ihttp_socket.hpp>
+#include <logger.hpp>
+
+#include <boost/asio.hpp>
+#include <boost/asio/experimental/awaitable_operators.hpp>
+#include <boost/beast.hpp>
+#include <boost/system/error_code.hpp>
+
+#include <exception>
+#include <string>
+
+namespace http_client
+{
+    void HttpSocket::CallAsyncConnect(const boost::asio::ip::tcp::resolver::results_type& endpoints,
+                                      boost::system::error_code& ec,
+                                      bool& connectionSuccess)
+    {
+        boost::asio::async_connect(m_socket,
+                                   endpoints,
+                                   [&](const boost::system::error_code& errorCode, const auto&)
+                                   {
+                                       if (!errorCode)
+                                       {
+                                           connectionSuccess = true;
+                                           ec = errorCode;
+                                       }
+                                   });
+    }
+
+    void HttpSocket::Connect(boost::asio::io_context& ioContext,
+                             const boost::asio::ip::tcp::resolver::results_type& endpoints,
+                             boost::system::error_code& ec,
+                             const std::chrono::seconds timeOut)
+    {
+        try
+        {
+            bool connectionSuccess = false;
+
+            CallAsyncConnect(endpoints, ec, connectionSuccess);
+
+            ioContext.run_for(timeOut); // Run for 2 seconds();
+            if (!connectionSuccess)
+            {
+                ec = boost::asio::error::timed_out;
+                LogDebug("Connection timed out");
+            }
+        }
+        catch (const std::exception& e)
+        {
+            LogDebug("Exception thrown: {}", e.what());
+        }
+    }
+
+    boost::asio::awaitable<void>
+    HttpSocket::AsyncConnect(const boost::asio::ip::tcp::resolver::results_type& endpoints, // NOLINT
+                             boost::system::error_code& ec,                                 // NOLINT
+                             const std::chrono::seconds timeOut)
```

Comment:
Please specify which warning is being suppresed as mentioned in other comments.

Reply:
Done